rclrs/src/wait.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 DCS Corporation, All Rights Reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// DISTRIBUTION A. Approved for public release; distribution unlimited.
+// OPSEC #4584.
+
+use std::borrow::BorrowMut;
+use std::cell::{Ref, RefCell, RefMut};
+use std::rc::{Rc, Weak};
+
+use crate::{SubscriptionBase, error::*};
+use crate::Handle;
+
+use crate::rcl_bindings::*;
+
+use anyhow::{Context, Error, Result};
+use rclrs_common::error::WaitSetError;
+
+pub struct WaitSet {
+    pub wait_set: rcl_wait_set_t,
+    initialized: bool,
+}
+
+impl WaitSet {
+    /// Creates a new WaitSet object.
+    ///
+    /// Under the hood, this calls `rcl_get_zero_initialized_wait_set()`, and stores it
+    /// within the WaitSet struct, while also noting that the returned value is uninitialized.
+    pub fn new() -> Self {
```

Comment:
I'd fuse `new` and `init`, it makes sense to have the both in the same function because one wouldn't reinitialize a previously initialized waitset.